src/index.ts

```diff
@@ -0,0 +1,9 @@
+export function greet(
+  inputName: string
+): { message: string } | { error: string } {
```

Comment:
Great example of when to use a return type!

There's a concept in functional programming sometimes called `tagged union type` that you could use when writing code that returns objects that hold ok data or error data.

It would look something like this:
```ts
type GreetResult = {ok: true, message: string} | {ok: false, error: string };

function greet(name: string): GreetResult { }

const greetResult = greet("")

if(!greetResult.ok) {
  console.error(greetResult.error)
  return;
}

console.log(greetResult.message)
```

The if-statement will narrow the type down so you can access `error` inside the if-statement, but not outside it.
PS: this is intermediary TypeScript stuff, so don't worry too much about it, but I think you are close to using it already, so might as well mention it :-)